model/task_queue_service_dependency.go

```diff
@@ -348,6 +348,20 @@ func (d *basicCachedDAGDispatcherImpl) FindNextTask(spec TaskSpec) *TaskQueueIte
 			if !dependenciesMet {
 				continue
 			}
+
+			// AMI Updated time is only provided if the host is running with an outdated AMI.
+			// If the task was created after the time that the AMI was updated, then we should wait for an updated host.
+			if !utility.IsZeroTime(amiUpdatedTime) && nextTaskFromDB.IngestTime.After(amiUpdatedTime) {
```

Comment:
Since this is based on ingest time, are restarted tasks expected to use the latest AMI or is it okay for them to use an older one?

Reply:
Yeah good question, restarted tasks would use an older one. I couldn't quite decide which was better, scheduled time or ingest time. I was thinking that scheduled tasks are a superset of ingest tasks, so more tasks would be affected by this logic, whereas ingest time requires that it be a new item. I think it might be good to limit what tasks hit this constraint given the cons but it's not our typical way of handling tasks.